src/Main.elm

```diff
@@ -136,61 +136,68 @@ loadRoute maybeRoute model =
     let
         session =
             toSession model
-    in
-    case maybeRoute of
-        Nothing ->
-            ( NotFound session, Cmd.none )
 
-        Just Route.Home ->
-            let
-                ( subModel, subMsg ) =
-                    Home.init session
-            in
-            ( Home subModel, Cmd.map GotHomeMsg subMsg )
-
-        Just (Route.Auth Nothing) ->
-            let
-                ( subModel, subMsg ) =
-                    Auth.init session
-            in
-            ( Auth subModel, Cmd.map GotAuthMsg subMsg )
-
-        Just (Route.Auth (Just jwt)) ->
-            let
-                ( subModel, subMsg ) =
-                    PagesSession.init session jwt
-            in
-            ( Session subModel, Cmd.map GotPagesSessionMsg subMsg )
-
-        Just Route.Logout ->
-            ( Logout session
-            , Cmd.batch
-                [ Session.logout
-                , Route.replaceUrl (Session.navKey session) Route.Home
-                ]
-            )
-
-        Just Route.Capture ->
-            let
-                ( subModel, subMsg ) =
-                    Capture.init session
-            in
-            ( Capture subModel, Cmd.map GotCaptureMsg subMsg )
-
-        Just (Route.CaptureTimers idCapture) ->
-            let
-                ( subModel, subMsg ) =
-                    CaptureTimers.init session idCapture
-            in
-            ( CaptureTimers subModel, Cmd.map GotCaptureTimersMsg subMsg )
+        privateRoute =
+            Maybe.withDefault False <| Maybe.map Route.isPrivate maybeRoute
+    in
+    -- redirect to login page if user attempt to access private page as guest
+    if Session.isGuest session && privateRoute then
```

Comment:
the `loadRoute` function is called each time the url changed. So this is a good place to check the session of the user and redirect to the login page if the route is "private".